src/schema/users.ts

```diff
@@ -1176,6 +1193,68 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return streak;
     },
+    streakRecover: async (
+      _,
+      __,
+      ctx: AuthContext,
+    ): Promise<StreakRecoverQueryResult> => {
+      const cantRecoverResult: StreakRecoverQueryResult = {
+        canDo: false,
+        cost: 0,
+        oldStreakLength: 0,
+      };
+
+      const { userId } = ctx;
+
+      const key = generateStorageKey(
+        StorageTopic.Streak,
+        StorageKey.Reset,
+        userId,
+      );
+
+      const oldStreakLength = Number(await getRedisObject(key));
+      const userDontHaveOldStreak =
+        !!oldStreakLength && !isNumber(oldStreakLength);
+      if (userDontHaveOldStreak) {
+        return cantRecoverResult;
+      }
+
+      const streak = await ctx.con.getRepository(UserStreak).findOneBy({
```

Comment:
This should always exist right?

Reply:
Don't know if every logged user must have a streak, if yes I can remove the if 👀

Reply:
Yes. We have a trigger to create a streak entity once user is inserted to the DB.

Reply:
Updated! ✅